experimental/plugins/lando-platformsh/lib/auth.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+// Modules
+const _ = require('lodash');
+
+// Stuff
+const getTokens = tokens => _(tokens).map(token => ({name: token.email, value: token.token})).value();
+
+// Helper to get pantheon auth non-interactive options
+const getInteractiveOptions = (tokens = []) => ({
+  'auth': {
+    interactive: {
+      choices: _.flatten([getTokens(tokens), [{name: 'add or refresh a token', value: 'more'}]]),
+      when: answers => !_.isEmpty(tokens),
+      weight: 100,
+    },
+  },
+  'api-token': {
+    hidden: true,
+    interactive: {
+      name: 'auth',
+      type: 'password',
+      message: 'Enter a Platform.sh API token',
+      when: answers => _.isEmpty(tokens) || (_.get(answers, 'auth', '') === 'more'),
+      weight: 101,
+    },
+  },
+});
+
+// Helper to get pantheon auth non-interactive options
+const getNonInteractiveOptions = (token, email) => ({
+  'auth': {
+    default: token,
+    defaultDescription: email,
+  },
+});
+
+/*
+ * Helper to build a pull command
+ */
+exports.getAuthOptions = ({email = false, token = false} = {}, tokens = []) => {
+  if (email && token) return getNonInteractiveOptions(token, email);
```

Comment:
@mikemilano do you have a specific example?

Reply:
@pirog `if (email && token) return getNonInteractiveOptions(token, email);` I was thinking it also makes it more difficult if someone wants to throw some debug in that condition since they'll have to reformat with brackets to debug, then remove them when they are done.

If you were talking about examples of projects, Symfony and Drupal off the top of my head.